frappe/model/document.py

```diff
@@ -8,7 +8,7 @@
 from frappe.utils import flt, cstr, now, get_datetime_str, file_lock, date_diff
 from frappe.model.base_document import BaseDocument, get_controller
 from six import iteritems, string_types
-from frappe.model.naming import set_new_name
+from frappe.model.naming import set_new_name, gen_new_name_for_cancelled_doc, validate_name
```

Comment:
`gen_new_name_for_cancelled_doc` is being used here but this method does not exist.

https://github.com/frappe/frappe/blob/18201b826423752c2d3c64274f28952c4b9b2347/frappe/model/document.py#L1364-L1368

Inconsistent backporting somewhere? Unsure of the issue